scripts/votca_compare.in

```diff
@@ -0,0 +1,141 @@
+VERSION='@PROJECT_VERSION@ #TOOLS_GIT_ID#'
+import sys
+import os
+import getpass
+import argparse
+
+PROGTITLE = 'THE VOTCA::TOOLS FILE COMPARISON'
+PROGDESCR = 'COMPARES TWO FILES TO EACH OTHER'
+VOTCAHEADER = '''\
+==================================================
+========   VOTCA (http://www.votca.org)   ========
+==================================================
+
+{progtitle}
+
+please submit bugs to bugs@votca.org 
+votca_compare, version {version}
+
+'''.format(version=VERSION, progtitle=PROGTITLE)
+
+def xxquit(what=''):
+  if what != '':
+    cprint.Error("ERROR: {what}".format(what=what))
+  sys.exit(1)
+
+def RepresentsFloat(s):
+    try: 
+        float(s)
+        return True
+    except ValueError:
+        return False 
+
+# =============================================================================
+# PROGRAM OPTIONS
+# =============================================================================
+
+class ToolsHelpFormatter(argparse.HelpFormatter):
+  def _format_usage(self, usage, action, group, prefix):
+    return VOTCAHEADER
+    
+progargs = argparse.ArgumentParser(prog='votca_compare',
+    formatter_class=lambda prog: ToolsHelpFormatter(prog,max_help_position=70),
+  description=PROGDESCR)
+  
+progargs.add_argument('-f1', '--file1',
+    dest='file1',   
+    action='store',
+    required=True,
+    type=str,
+    help='File1 to compare.')
+
+progargs.add_argument('-f2', '--file2',
+    dest='file2',   
+    action='store',
+    required=True,
+    type=str,
+    help='File2 to compare')
+
+progargs.add_argument('--etol',
+    dest='etol',   
+    action='store',
+    type=float,
+    default=1e-5,
+    help='Tolerance for floats to be regarded equal,default=1e-5')
+    
+OPTIONS = progargs.parse_args()
+
+# =============================================================================
+# Compare Execution
+# =============================================================================
+
+
+if not os.path.isfile(os.path.join(os.getcwd(),OPTIONS.file1)):
+  progargs.print_help()
+  xxquit("\nQuit here, because:{} is not a file(option -f1)".format(OPTIONS.file1))
+if not os.path.isfile(os.path.join(os.getcwd(),OPTIONS.file2)):
+  progargs.print_help()
+  xxquit("\nQuit here, because:{} is not a file(option -f2)".format(OPTIONS.file2))
+
+print ("Comparing {} and {} to a tolerance of {}".format(OPTIONS.file1,OPTIONS.file2,OPTIONS.etol))
+
+data1=[]
+with open(os.path.join(os.getcwd(),OPTIONS.file1),"r") as f1:
+  for line in f1:
+    if line[0]=="#":
+      continue;
+    else:
+      tok=line.split()
+      data1.append(tok)
+data2=[]
+
+with open(os.path.join(os.getcwd(),OPTIONS.file2),"r") as f2:
+  for line in f2:
+    if line[0]=="#":
+      continue;
+    else:
+      tok=line.split()
+      data2.append(tok)
+
+if(len(data2) != len(data1)):
+  print ("Files have different number of data points {} vs {}".format(len(data1),len(data2)))
+
+logs=[]
+
+for i,(d1,d2) in enumerate(zip(data1,data2)):
```

Comment:
In fact maybe this could be done easier, without reading the whole file first to the array.
However, we lose the comparison against the number of lines.

```python
with open(os.path.join(os.getcwd(), OPTIONS.file1), 'r') as f1, open(os.path.join(os.getcwd(), OPTIONS.file2), 'r') as f2:                                                              
     for i, (line1, line2) in enumerate(zip(f1, f2)):                                             
         if line1.startswith('#') or line2.startswith('#'):                                       
             continue                                                                             
         if j, (entry1, entry2) in enumerate(zip(line1.split(), line2.split())):                  
             if RepresentsFloat(entry1) and RepresentsFloat(entry2):                                              
                 if abs(float(entry1) - float(entry2)) > OPTIONS.etol:
                     logs.append([i, j, entry1, entry2])  
             else entry1 != entry2:        
                 logs.append([i, j, entry1, entry2])          
```